File: Studio2015/Win32Lib/Person.cpp
#include "stdafx.h"
#include "Person.h"


Person::Person() : IPerson()
{
}

Person::Person(std::string name, int age) : IPerson(name,  age)
{
}

Person::~Person()
{
}

std::string Person::GetName()
{
	return _name;
}

void Person::SetName(std::string name)
{
	_name = name;
}

int Person::GetAge()
{
	return _age;
}

void Person::SetAge(int age)
{
	_age = age;
}

void Person::GetInfo(std::string & name, int & age)
{
	name = _name;
	age = _age;
}

void Person::SetInfo(std::string name, int age)
{
	_name = name;
	_age = age;
}

File: Studio2015/CSharpLib/Person.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace CSharpLib
{
    public class Person : IPerson
    {
        public Person(string name, int age)
        {
            Name = name;
            Age = age;
        }

        public int Age { get; set; }

        public string Name { get; set; }

        public void GetInfo(out int age, ref string info)
        {
            throw new NotImplementedException();
        }
        public void GetInfo(out string info)
        {
            throw new NotImplementedException();
        }
        public void GetInfo(string name, out int age, ref string info)
        {
            throw new NotImplementedException();
        }

        public bool GetInfo()
        {
            throw new NotImplementedException();
        }
    }
}
